pyccel/parser/semantic.py

```diff
@@ -474,22 +471,20 @@ def _infere_type(self, expr, **settings):
 
             var = expr.internal_var
 
+            d_var['memory_handling'] = 'alias' if isinstance(var, Variable) else var.memory_handling
```

Comment:
This didn't make sense before your PR, but since I've noticed it would be nice if we could fix it. I'm not sure there will be a `var.memory_handling` available if `var` is not a `Variable` (it will probably be something like an arithmetic operator). I think this should default to `heap`

Reply:
Only the `else` part or the whole expression?